cli/manage.go

```diff
@@ -128,34 +130,56 @@ func setupReplication(c *cli.Context, cluster cluster.Cluster, server *api.Serve
 	primary := api.NewPrimary(cluster, tlsConfig, &statusHandler{cluster, candidate, follower}, c.Bool("cors"))
 	replica := api.NewReplica(primary, tlsConfig)
 
-	go func() {
-		candidate.RunForElection()
-		electedCh := candidate.ElectedCh()
-		for isElected := range electedCh {
+	go run(candidate, server, primary, replica)
+	go follow(follower, replica, addr)
+
+	server.SetHandler(primary)
+}
+
+func run(candidate *leadership.Candidate, server *api.Server, primary *mux.Router, replica *api.Replica) {
+	electedCh, errCh := candidate.RunForElection()
+	for {
+		select {
+		case isElected := <-electedCh:
 			if isElected {
-				log.Info("Cluster leadership acquired")
+				log.Info("Leader Election: Cluster leadership acquired")
 				server.SetHandler(primary)
 			} else {
-				log.Info("Cluster leadership lost")
+				log.Info("Leader Election: Cluster leadership lost")
 				server.SetHandler(replica)
 			}
+
+		case err := <-errCh:
+			log.Error(err)
+			time.Sleep(defaultRecoverTime)
+			go run(candidate, server, primary, replica)
```

Comment:
This is a little weird - when you call `run()` it will block but then when it returns it will continue running as a goroutine in the background.

How about just returning here? And in `setupReplication`:
```go
go func() {
  for {
    run(...)
    time.Sleep(...)
   }
}
```